tests/test_fetch.py

```diff
@@ -201,79 +209,208 @@ def test_number_outputfiles(capsys):
+    producttype = era5._product_type()
+    assert producttype == 'monthly_averaged_ensemble_members_by_hour_of_day'
+
+    era5.synoptic = False
+    producttype = era5._product_type()
+    assert producttype == 'monthly_averaged_ensemble_members'
+
+    # Preliminary back extension monthly data have different names
     era5.prelimbe = True
     producttype = era5._product_type()
-    assert producttype == 'reanalysis-monthly-means-of-daily-means'
+    assert producttype == 'members-monthly-means-of-daily-means'
 
-    era5.prelimbe = False
     era5.synoptic = True
     producttype = era5._product_type()
-    assert producttype == 'monthly_averaged_reanalysis_by_hour_of_day'
+    assert producttype == 'members-synoptic-monthly-means'
 
-    era5.prelimbe = True
+    era5.ensemble = False
     producttype = era5._product_type()
     assert producttype == 'reanalysis-synoptic-monthly-means'
 
-    era5.ensemble = True
+    era5.synoptic = False
     producttype = era5._product_type()
-    assert producttype == 'members-synoptic-monthly-means'
+    assert producttype == 'reanalysis-monthly-means-of-daily-means'
 
+    # ERA5 land has more limited options
+    era5.land = True
     era5.prelimbe = False
-    era5.ensemble = False
-    era5.statistics = True
+    producttype = era5._product_type()
+    assert producttype == 'monthly_averaged_reanalysis'
+
+    era5.synoptic = True
     producttype = era5._product_type()
     assert producttype == 'monthly_averaged_reanalysis_by_hour_of_day'
 
+    era5.period = 'hourly'
+    producttype = era5._product_type()
+    assert producttype is None
+
+
+def test_check_levels():
+    """Test _check_levels function of Fetch class"""
+    era5 = initialize()
+    era5.variables = "temperature"
+
+    # No levels should raise
+    with pytest.raises(ValueError):
+        era5._check_levels()
+
+    # Valid levels should pass
+    era5.pressure_levels = [1000, 950]
+    era5._check_levels()
+
+    # Invalid levels should raise
+    era5.pressure_levels = [777]
+    with pytest.raises(ValueError):
+        era5._check_levels()
+
+
+def test_check_variable():
+    """Test _check_variable function of Fetch class."""
+    era5 = initialize()
+
+    land_only_variable = "snow_cover"
+    slev_only_variable = "vertical_integral_of_mass_tendency"
+    missing_monthly_var = "altimeter_wave_height"
+
+    # Invalid variables should raise
+    with pytest.raises(ValueError):
+        era5._check_variable("invalid_precipitation")
+
+    # Valid variable with invalid dataset option should raise
+    with pytest.raises(ValueError):
+        era5._check_variable(land_only_variable)
+
+    # But with the right dataset option it should pass
+    era5.land = True
+    era5._check_variable(land_only_variable)
+
+    # And variables incompatible with ERA5 land should fail
+    with pytest.raises(ValueError):
+        era5._check_variable(slev_only_variable)
+
+    # Missing monthly vars should pass if period is hourly
```

Comment:
perhaps set the period explicitly here? I think this currently depends on the default period setting, right?